src/mono/wasm/debugger/DebuggerTestSuite/Inspector.cs

```diff
@@ -158,14 +159,22 @@ async Task OnMessage(string method, JObject args, CancellationToken token)
                 case "Runtime.consoleAPICalled":
                     _logger.LogInformation(FormatConsoleAPICalled(args));
                     break;
+                case "Inspector.detached":
+                case "Inspector.targetCrashed":
+                case "Inspector.targetReloadedAfterCrash":
+                    fail = true;
+                    break;
+                case "Runtime.exceptionThrown":
+                    _logger.LogDebug($"Failing all waiters because: {method}: {args}");
+                    fail = true;
+                    break;
             }
             if (eventListeners.TryGetValue(method, out var listener))
             {
                 await listener(args, token).ConfigureAwait(false);
             }
-            else if (String.Compare(method, "Runtime.exceptionThrown") == 0)
+            else if (fail)
             {
-                _logger.LogDebug($"Failing all waiters because: {method}: {args}");
                 FailAllWaiters(new ArgumentException(args.ToString()));
```

Comment:
We could instead pass `new ArgumentException ($"{method}: {args}")` to `FailAllWaiters` here. It will always get surfaced then, and we don't need to do the `LogDebug` either.

And we want to do that for all 4 cases where it's crashing, or exception is thrown etc.

Reply:
There are places which parse back the Json. See https://github.com/pavelsavara/runtime/blob/pr_wasm_debugger_crash/src/mono/wasm/debugger/DebuggerTestSuite/Tests.cs#L57